src/platform/nrfconnect/crypto/KMUSessionKeystore.cpp

```diff
@@ -0,0 +1,202 @@
+    KMUHmacKeyAttributes() :
+        KeyAttributesBase(PSA_KEY_TYPE_HMAC, PSA_ALG_HMAC(PSA_ALG_SHA_256), PSA_KEY_USAGE_SIGN_MESSAGE | PSA_KEY_USAGE_EXPORT,
+                          CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES * 8)
+    {}
+};
+} // namespace
+
+// TODO: Remove this implementation once the Cracen supports the base AesKeyAttributes implementation from PSASessionKeystore
+CHIP_ERROR KMUSessionKeystore::CreateKey(const Symmetric128BitsKeyByteArray & keyMaterial, Aes128KeyHandle & key)
+{
+    // Destroy the old key if already allocated
+    DestroyKey(key);
+
+    KMUAesKeyAttributes attrs;
+    psa_status_t status =
+        psa_import_key(&attrs.Get(), keyMaterial, sizeof(Symmetric128BitsKeyByteArray), &key.AsMutable<psa_key_id_t>());
+    LogPsaError(status);
+    VerifyOrReturnError(status == PSA_SUCCESS, CHIP_ERROR_INTERNAL);
+
+    return CHIP_NO_ERROR;
+}
+
+// TODO: Remove this implementation once the Cracen supports the base HmacKeyAttributes implementation from PSASessionKeystore
+CHIP_ERROR KMUSessionKeystore::CreateKey(const Symmetric128BitsKeyByteArray & keyMaterial, Hmac128KeyHandle & key)
+{
+    // Destroy the old key if already allocated
+    DestroyKey(key);
+
+    KMUHmacKeyAttributes attrs;
+    psa_status_t status =
+        psa_import_key(&attrs.Get(), keyMaterial, sizeof(Symmetric128BitsKeyByteArray), &key.AsMutable<psa_key_id_t>());
+    LogPsaError(status);
+    VerifyOrReturnError(status == PSA_SUCCESS, CHIP_ERROR_INTERNAL);
+
+    return CHIP_NO_ERROR;
+}
+
+// TODO: Remove this implementation once the Cracen supports the base AesKeyAttributes implementation from PSASessionKeystore
+CHIP_ERROR KMUSessionKeystore::DeriveKey(const P256ECDHDerivedSecret & secret, const ByteSpan & salt, const ByteSpan & info,
+                                         Aes128KeyHandle & key)
+{
+    PsaKdf kdf;
+    ReturnErrorOnFailure(kdf.Init(secret.Span(), salt, info));
+
+    KMUAesKeyAttributes attrs;
+
+    return kdf.DeriveKey(attrs.Get(), key.AsMutable<psa_key_id_t>());
+}
+
+CHIP_ERROR KMUSessionKeystore::DeriveSessionKeys(PsaKdf & kdf, Aes128KeyHandle & i2rKey, Aes128KeyHandle & r2iKey,
+                                                 AttestationChallenge & attestationChallenge)
+{
+    CHIP_ERROR error;
+    KMUAesKeyAttributes attrs;
+
+    SuccessOrExit(error = kdf.DeriveKey(attrs.Get(), i2rKey.AsMutable<psa_key_id_t>()));
+    SuccessOrExit(error = kdf.DeriveKey(attrs.Get(), r2iKey.AsMutable<psa_key_id_t>()));
+    SuccessOrExit(error = kdf.DeriveBytes(MutableByteSpan(attestationChallenge.Bytes(), AttestationChallenge::Capacity())));
+
+exit:
+    if (error != CHIP_NO_ERROR)
+    {
+        DestroyKey(i2rKey);
+        DestroyKey(r2iKey);
+    }
+
+    return error;
+}
+
+#if CHIP_CONFIG_ENABLE_ICD_CIP
+CHIP_ERROR KMUSessionKeystore::PersistICDKey(Symmetric128BitsKeyHandle & key)
+{
+    CHIP_ERROR err;
+    psa_key_id_t newKeyId = PSA_KEY_ID_NULL;
+    psa_key_attributes_t attrs;
+    psa_status_t s = PSA_SUCCESS;
```

Comment:
Is it used anywhere?

Reply:
no, it was for testing purposes, will remove it.